### ISSUE TITLE
Adding joins to relations

### ISSUE THREAD

**Issue by pnomolos** [vlucas/Spot#18]
I'd like the ability to add join tables to relation so that you could do things such as ordering by a field on the joined table.  What are your thoughts on something like this?

**Comment by vlucas** [comment]
You know I'm going to ask you for a use case :). I have not needed this myself, so I am especially curious now!

**Comment by pnomolos** [comment]
Sure :)

Devices have and belong to many Categories.  Categories belong to Types.  Types are ordered.  I'd like to order the Categories by the order of whatever Types they're attached to.

**Comment by vlucas** [comment]
And the categories have no other criteria on which they are ordered? Typically a join table doesn't have any relevant information beyond the foreign key ID's.

**Comment by pnomolos** [comment]
That's correct.  The join table is between Devices and Categories - Categories is not a join table and doesn't have any specific order information.
